packages/gatsby/src/utils/adapter/manager.ts

```diff
@@ -458,17 +458,28 @@ function getRoutesManifest(): RoutesManifest {
 
   // redirect routes
   for (const redirect of state.redirects.values()) {
+    const {
+      fromPath,
+      toPath,
+      statusCode,
+      isPermanent,
+      ignoreCase,
+      redirectInBrowser,
+      ...platformSpecificFields
+    } = redirect
+
     addRoute({
-      path: redirect.fromPath,
+      path: fromPath,
       type: `redirect`,
-      toPath: redirect.toPath,
+      toPath: toPath,
       status:
-        redirect.statusCode ??
-        (redirect.isPermanent
+        statusCode ??
+        (isPermanent
           ? HTTP_STATUS_CODE.MOVED_PERMANENTLY_301
           : HTTP_STATUS_CODE.FOUND_302),
-      ignoreCase: redirect.ignoreCase,
+      ignoreCase: ignoreCase,
       headers: BASE_HEADERS,
+      platformSpecificFields,
```

Comment:
```suggestion
      ...platformSpecificFields,
```
otherwise this would not really match this (public) type https://github.com/gatsbyjs/gatsby/blob/5dbcf9ea956880af7541aaf33ed2f764ec8632d5/packages/gatsby/src/utils/adapter/types.ts#L47-L64

(technically it does matches it, but this is more that shape of redirect is more about having platform specific fields as top level fields and not have actual `platformSpecificFields` field that is object with platform specific fields. 

Also with this as-is the current code in this PR in packages/gatsby-adapter-netlify/src/route-handler.ts wouldn't actually work as it does expect `force` field to be top-level field and not nested under `platformSpecificFields`

This change will need some adjustment in test added to packages/gatsby/src/utils/adapter/__tests__/manager.ts